community/detectors/weblogic_cve_2020_14882/src/test/java/com/google/tsunami/plugins/detectors/cves/cve202014882/Cve202014882VulnDetectorTest.java

```diff
@@ -0,0 +1,149 @@
+        TargetInfo.newBuilder()
+            .addNetworkEndpoints(forHostname(mockWebServer.getHostName()))
+            .build();
+
+    DetectionReportList detectionReports = detector.detect(targetInfo, ImmutableList.of(service));
+
+    assertThat(detectionReports.getDetectionReportsList())
+        .containsExactly(
+            DetectionReport.newBuilder()
+                .setTargetInfo(targetInfo)
+                .setNetworkService(service)
+                .setDetectionTimestamp(
+                    Timestamps.fromMillis(Instant.now(fakeUtcClock).toEpochMilli()))
+                .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+                .setVulnerability(
+                    Vulnerability.newBuilder()
+                        .setMainId(
+                            VulnerabilityId.newBuilder()
+                                .setPublisher("TSUNAMI_COMMUNITY")
+                                .setValue("CVE_2020_14882"))
+                        .setSeverity(Severity.CRITICAL)
+                        .setTitle("CVE-2020-14882: Weblogic management console permission bypass")
+                        .setDescription(
+                            "Vulnerability in the Oracle WebLogic Server product of Oracle Fusion Middleware (component: Console). "
+                                + "Supported versions that are affected are 10.3.6.0.0, 12.1.3.0.0, 12.2.1.3.0, 12.2.1.4.0 and 14.1.1.0.0. "
+                                + "Easily exploitable vulnerability allows unauthenticated attacker with network access via HTTP to "
+                                + "compromise Oracle WebLogic Server. Successful attacks of this vulnerability can result in takeover of "
+                                + "Oracle WebLogic Server")
+                        .setRecommendation(
+                            "Go to the oracle official website to download the latest weblogic patch."))
+                .build());
+  }
+
+  @Test
+  public void detect_whenNotVulnerable_returnsNoVulnerability() throws IOException {
+    mockWebResponse("Hello World");
+    ImmutableList<NetworkService> httpServices =
+        ImmutableList.of(
+            NetworkService.newBuilder()
+                .setNetworkEndpoint(
+                    forHostnameAndPort(mockWebServer.getHostName(), mockWebServer.getPort()))
+                .setTransportProtocol(TransportProtocol.TCP)
+                .setServiceName("http")
+                .build());
+    TargetInfo targetInfo =
+        TargetInfo.newBuilder()
+            .addNetworkEndpoints(forHostname(mockWebServer.getHostName()))
+            .build();
+
+    DetectionReportList detectionReports = detector.detect(targetInfo, httpServices);
+
+    assertThat(detectionReports.getDetectionReportsList()).isEmpty();
+  }
+
+  private void mockWebResponse(String body) throws IOException {
+    mockWebServer.start();
+    mockWebServer.enqueue(new MockResponse().setResponseCode(200).setBody(body));
```

Comment:
The detector has a logic to handle `302` redirect with the `Set-Cookie` header. Can you please add a unit test for this case as well?